### PR TITLE
Add tests for UMS PositionPointer overflowing

### PR DESCRIPTION
**Pull request by ianhays**
Adds some tests for https://github.com/dotnet/coreclr/pull/11376 which resolved https://github.com/dotnet/coreclr/issues/11199.

Requires a new version of coreclr for the tests to pass.

cc: @jkotas

### CODE REVIEW

**Comment by jkotas** [comment]
BTW: An existing UMS test is failing because of the fix in the CoreCLR update job: https://ci.dot.net/job/dotnet_corefx/job/master/job/windows_nt_debug_prtest/7514/testReport/junit/System.IO.Tests/UmsWriteTests/CannotWriteWithOverflow/

File: src/System.IO.UnmanagedMemoryStream/tests/UmsSecurityTest.cs [hunk 1]
```diff
@@ -27,6 +27,8 @@ public static void ChangePositionViaPointer()
                         stream.PositionPointer = stream.PositionPointer - 1;
                     });
 
+                    Assert.Throws<ArgumentOutOfRangeException>(() => stream.PositionPointer = (byte*)ulong.MaxValue);
```
Comment (jkotas) [inline src/System.IO.UnmanagedMemoryStream/tests/UmsSecurityTest.cs]: This is fragile. It the OS allocates `data` at very high address (e.g. `ulong.MaxValue - 0x10000000`), this test will fail. You may want to move this check into the OverflowPositionPointer test where the stream is initialized with a predictable base address.

**Comment by ianhays** [comment]
>BTW: An existing UMS test is failing because of the fix in the CoreCLR update job: 

Thanks for notifying me. I'm testing 64 bit so I missed that one. I'll remove the AOoRE expectation.

File: src/System.IO.UnmanagedMemoryStream/tests/UmsSecurityTest.cs (hunk 1 above)
Comment (ianhays) [inline src/System.IO.UnmanagedMemoryStream/tests/UmsSecurityTest.cs]: Updated.

File: src/System.IO.UnmanagedMemoryStream/tests/UmsWrite.cs [hunk 2]
```diff
@@ -111,7 +111,7 @@ public static void CannotWriteWithOverflow()
 
                 if (IntPtr.Size == 4)
```
Comment (jkotas) [inline src/System.IO.UnmanagedMemoryStream/tests/UmsWrite.cs]: We should remove this whole `if (IntPtr.Size == 4)` path. The other path should work on all platforms now.

File: src/System.IO.UnmanagedMemoryStream/tests/UmsSecurityTest.cs [hunk 3]
```diff
@@ -38,6 +38,21 @@ public static void ChangePositionViaPointer()
         }
     }
 
+    [Fact]
+    public static void OverflowPositionPointer()
```
Comment (jkotas) [inline src/System.IO.UnmanagedMemoryStream/tests/UmsSecurityTest.cs]: This test won't work on netfx - needs attribute to disable it.

**Comment by stephentoub** [comment]
Cherry-picked into https://github.com/dotnet/corefx/pull/19440

**Comment by ianhays** [comment]
Thanks @stphentoub! I'll close this PR out.

**Pull request closed without merging**